src/pyodide/internal/_workers.py

```diff
@@ -930,6 +931,50 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+class _WorkflowStepWrapper:
+    def __init__(self, js_step):
+        self._js_step = js_step
+
+    def do(self, name, callback=None, config=None):
+        if callback:
+
+            async def wrapper():
+                return _do_call(self, name, config, callback)
+
+            return wrapper()
+
+        # Return a decorator that will execute the function when called
+        def decorator(func):
+            @functools.wraps(func)
```

Comment:
We should get rid of `@functools.wraps()` since it is really for decorators that don't change the arguments:
```suggestion
```